blas/gonum/level3double.go

```diff
@@ -498,7 +516,11 @@ func (Implementation) Dsyrk(ul blas.Uplo, tA blas.Transpose, n, k int, alpha flo
 	}
 	for i := 0; i < n; i++ {
 		ctmp := c[i*ldc : i*ldc+i+1]
-		if beta != 0 {
```

Comment:
The actual bug was here, the comparison `beta != 0` should have been `beta != 1`. I added the 0 special case (optimization) together with a fix, and for consistency I added it also to the other paths, all lumped into one commit. I'm sorry for taking this shortcut, I can still split it.

Reply:
No need to split it, but can you add a note to the commit message explaining the issue and what was done?

Reply:
I split it anyway, I think it's clearer this way. Also added a more descriptive message to the fixing commit.